package/MDAnalysis/analysis/rms.py

```diff
@@ -535,29 +541,77 @@ def __init__(self, atomgroup, reference=None, select='all',
                         igroup, sel['reference'], sel['mobile'],
                         len(atoms['reference']), len(atoms['mobile'])))
 
-        # Explicitly check for "mass" because this option CAN
-        # be used with groupselection. (get_weights() returns the mass array
-        # for "mass")
-        if not iterable(self.weights) and self.weights == "mass":
+
+        # check weights type
+
+        if self.weights == 'mass':
             pass
-        else:
+        elif self.weights is None:
+            pass
+        elif (np.array(self.weights).ndim == 1) & (np.array(self.weights).dtype in (np.dtype('float64'),np.dtype('int64'))):
             self.weights = get_weights(self.mobile_atoms, self.weights)
-
-        # cannot use arbitrary weight array (for superposition) with
-        # groupselections because arrays will not match
-        if (len(self.groupselections) > 0 and (
-                iterable(self.weights) or self.weights not in ("mass", None))):
-            raise ValueError("groupselections can only be combined with "
-                             "weights=None or weights='mass', not a weight "
-                             "array.")
+        elif not iterable(self.weights):
+            self.weights = get_weights(self.mobile_atoms, self.weights)
+        elif len(self.weights) != len(self.groupselections) + 1:
+            raise ValueError("Length of array of weights is not equal to length of groupselections + 1")
+        else:
+            weights_list = self.weights
+            for weights in weights_list:
+                if weights == 'mass':
+                    pass
+                elif weights is None:
+                    pass
+                elif (np.array(weights).ndim == 1) & (np.array(weights).dtype in (np.dtype('float64'),np.dtype('int64'))):
+                    # TODO:
+                    # check each length of weights matches the length of each group selection?
+                    pass
+
+                elif not iterable(self.weights):
+                    self.weights = get_weights(self.mobile_atoms, self.weights)
+                else:
+                    raise ValueError("Each groupselection can only be combined with "
+                                     "weight None or 'mass' or 1D float array")
+      
 
     def _prepare(self):
         self._n_atoms = self.mobile_atoms.n_atoms
+        # apply 'mass' weights for all selections
+        if self.weights == 'mass':
+           self.weights = ['mass'] * (len(self.groupselections) + 1)
+        # apply 'None' weights for all selections
```

Comment:
put the comment inside its block instead of before the conditional (here and elsewhere) – that makes the code more readable

Reply:
Note: If `self.weights` is an array you will get a `FutureWarning` and eventually this will become an error. Should be changed to work with both scalars and arrays.